packages/Checkbox/src/Checkbox.js

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import PropTypes from "prop-types";
+import uuid from "uuid/v4";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import CheckIcon from "@paprika/icon/lib/Check";
+import DashIcon from "@paprika/icon/lib/Dash";
+import checkboxStyles from "./Checkbox.styles";
+
+export const checkboxStates = {
+  CHECKED: "checked",
+  UNCHECKED: "unchecked",
+  INDETERMINATE: "indeterminate",
+};
+
+const propTypes = {
+  a11yText: PropTypes.string,
+  children: PropTypes.node,
+  isDisabled: PropTypes.bool,
+  onChange: PropTypes.func.isRequired,
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+  checkedState: PropTypes.oneOf(Object.values(checkboxStates)),
+};
+
+const defaultProps = {
+  a11yText: null,
+  children: null,
+  isDisabled: false,
+  size: ShirtSizes.MEDIUM,
+  checkedState: checkboxStates.UNCHECKED,
+};
+
+const Checkbox = props => {
+  const checkboxId = React.useRef(uuid()).current;
+  const inputRef = React.useRef(null);
+
+  const { a11yText, children, isDisabled, checkedState, size, ...moreProps } = props;
+
+  const styleProps = {
+    hasChildren: !!children,
+    size,
+  };
+
+  React.useEffect(() => {
+    if (inputRef.current && checkedState === checkboxStates.INDETERMINATE) {
+      inputRef.current.indeterminate = checkboxStates.INDETERMINATE;
+    }
+  }, [checkedState]);
+
+  const inputProps = {};
+  if (a11yText) inputProps["aria-label"] = a11yText;
+
+  return (
+    <div data-pka-anchor="checkbox" css={checkboxStyles} {...styleProps} {...moreProps}>
+      <input
+        type="checkbox"
+        id={checkboxId}
+        checked={checkedState === checkboxStates.CHECKED}
+        disabled={isDisabled}
+        ref={inputRef}
+        {...inputProps}
+      />
+      <label htmlFor={checkboxId}>
+        {children}
+        <CheckIcon className="checkbox-icon" aria-hidden data-pka-anchor="checkbox.icon.check" />
+        <DashIcon aria-hidden className="checkbox-icon" data-pka-anchor="checkbox.icon.indeterminate" />
```

Comment:
Oh, probably the don't need the `classNames`.

Reply:
I am using the classnames for some styling of the icons...

Reply:
Let's use the `pka-anchors` instead (in the next PR for the `<Radio>`)

Reply:
At least in the paprika components i think we should keep the testing attributes separate from styling purposes. (Consuming applications may use the testing attributes for styling though)

Discussed with @mikrotron, agreed that when I building out the radio input component, we should refactor the styles applied on these icon components to be variables coming from the `checkbox.styles.js`.